test/functional/blockprod_generate_pos_blocks_rand_genesis_keys.py

```diff
@@ -0,0 +1,438 @@
+            "key": {
+                "Schnorrkel": {
+                    "key": self.hex_to_dec_array(vrf_public_key),
+                },
+            },
+        }
+
+    def run_test(self):
+        global genesis_stake_private_key_hex, genesis_stake_public_key_hex
+        genesis_stake_private_key = self.stake_private_key(genesis_stake_private_key_hex[2:])
+        genesis_stake_public_key = self.stake_public_key(genesis_stake_public_key_hex[2:])
+
+        global genesis_vrf_private_key_hex, genesis_vrf_public_key_hex
+        genesis_vrf_private_key = self.vrf_private_key(genesis_vrf_private_key_hex[2:])
+        genesis_vrf_public_key = self.vrf_public_key(genesis_vrf_public_key_hex[2:])
+
+        #
+        # Transfer Genesis UTXO to AnyoneCanSpend
+        #
+
+        block_input_data = block_input_data_obj.encode({
+            "PoS": {
+                "stake_private_key": genesis_stake_private_key,
+                "vrf_private_key": genesis_vrf_private_key,
+                "pool_id": genesis_pool_id,
+                "kernel_inputs": [
+                        {
+                            "Utxo": {
+                                "id": {
+                                    "BlockReward": self.previous_block_id()
+                                },
+                                "index": 1,
+                            },
+                        },
+                ],
+                 "kernel_input_utxo": [
+                    {
+                        "CreateStakePool": [
+                            genesis_pool_id,
+                            {
+                                "value": "4000000000000000",
+                                "staker": {
+                                    "PublicKey": genesis_stake_public_key,
+                                },
+                                "vrf_public_key": genesis_vrf_public_key,
+                                "decommission_key": {
+                                    "PublicKey": genesis_stake_public_key,
+                                },
+                                "margin_ratio_per_thousand": 1000,
+                                "cost_per_block" : "0"
+                            },
+                        ],
+                    }
+                ],
+            }
+        }).to_hex()[2:]
+
+        transfer_transaction = {
+            "transaction": {
+                "version": 1,
+                "flags": 0,
+                "inputs": [
+                    {
+                        "Utxo": {
+                            "id": {
+                                "BlockReward": self.previous_block_id()
+                            },
+                            "index": 0,
+                        },
+                    }
+                ],
+                "outputs": [
+                    {
+                        "Transfer": [
+                            {
+                                "Coin": "10000000000000000",
```

Comment:
Nitpick: I think you can enter a number here instead of a string, and use the underscore syntax for better readability. Like `10_000_000_000_000_000`. Not sure it will fit into Python's numeric type but I suspect it uses some sort of bignum by default?

Reply:
While I don't disagree there, the problem is that the number of decimals isn't a multiple of three, so this won't help to make the number more understandable. We should use proper COIN factor, being 10^11 or whatever correct decimal count, then we have 10_000*COIN.

Reply:
Using `COIN` a 10^11, this is now fixed.